.lintrunner.toml

```diff
@@ -557,7 +557,7 @@ exclude_patterns = [
 command = [
     'python3',
     'tools/linter/adapters/grep_linter.py',
-    '--pattern=#include <pybind11\/',
+    '--pattern=#include <pybind11\/(^|[^(gil\.h)])',
```

Comment:
Why this change is needed?

Reply:
@malfet The `cpp_wrapper` code doesn't depend on the full `pybind11` library; it only needs the GIL handling classes. This linter ensures that we include all our `pybind11` class specializations when including any `pybind11` header, but those specializations aren't relevant to the GIL handling header specifically. Additionally, including the specializations would include the entire `pybind11` header, which is what we're trying to avoid here (for compile-time reasons).

This change just adds an exception for including the GIL header specifically.